genext2fs.c

```diff
@@ -2766,17 +2766,17 @@ add2fs_from_dir(filesystem *fs, uint32 this_nod, int squash_uids, int squash_per
 	uint32 uid, gid, mode, ctime, mtime;
 	const char *name;
 	FILE *fh;
-	DIR *dh;
-	struct dirent *dent;
+	struct dirent **dents = NULL;
 	struct stat st;
 	char *lnk;
-	uint32 save_nod;
+	uint32 save_nod, numdirs, i;
 	off_t filesize;
 
-	if(!(dh = opendir(".")))
+	if((numdirs = scandir(".", &dents, NULL, alphasort)) == -1)
```

Comment:
`alphasort` is locale aware. You need to set `LC_COLLATE` to a fixed value like `C` to produce the same output independent of the user's locale

Reply:
The man page states that this is already the default behavior. Would you rather make it explicit?
```
...
On startup of the main program, the portable "C" locale is selected as default.
...
```

Reply:
you are right!